parasolr/query.py

```diff
@@ -123,6 +127,22 @@ def count(self) -> int:
         query_opts['rows'] = 0
         return self.solr.query(**query_opts, wrap=False)['response']['numFound']
 
+    def facet_fields(self) -> dict:
+        """Return a dictionary of facets and their values/counts (also as a
+        dicitionary).
+        """
+        if self._result_cache is not None:
+            # wrap to process facets and return as dictionary
+            # for Django template support
+            qr = QueryResponse(self._result_cache)
+            return dict(qr.facet_counts.facet_fields)
```

Comment:
Why is this casting the return to a `dict`?

Reply:
This is to drag it out of being an `AttrDict` (they're recursive for Dict-like collections). I'm feeling increasingly less convinced about the utility of `AttrDict` given its issues with Django templating... but it does read more easily.